llvm/tools/llvm-objdump/llvm-objdump.cpp

```diff
@@ -2392,14 +2393,18 @@ disassembleObject(ObjectFile &Obj, const ObjectFile &DbgObj,
                     [=](const std::pair<uint64_t, SectionRef> &O) {
                       return O.first <= Target;
                     });
-                uint64_t TargetSecAddr = 0;
+                uint64_t TargetSecAddr = It == SectionAddresses.end() ? 0 : It->first;
+                bool FoundSymbols = false;
                 while (It != SectionAddresses.begin()) {
                   --It;
-                  if (TargetSecAddr == 0)
+                  if (It->first != TargetSecAddr) {
+                    if (FoundSymbols)
+                      break;
                     TargetSecAddr = It->first;
-                  if (It->first != TargetSecAddr)
-                    break;
+                  }
                   TargetSectionSymbols.push_back(&AllSymbols[It->second]);
+                  if (!AllSymbols[It->second].empty())
```

Comment:
Can we avoid a second `AllSymbols[It->second]` here since we already did it for the previous line. That's a map lookup, we shouldn't do it twice.